### PR TITLE
Alpaka implementation of FCS based on the dev/stdpar branch

### PR DESCRIPTION
**Pull request by tsulaiav**

### CODE REVIEW

File: FastCaloSimAnalyzer/FastCaloGpu/src/CaloGpuGeneral_al.cxx [hunk 1]
```diff
@@ -0,0 +1,198 @@
+/*
+  Copyright (C) 2002-2021 CERN for the benefit of the ATLAS collaboration
```
Comment (cgleggett) [inline FastCaloSimAnalyzer/FastCaloGpu/src/CaloGpuGeneral_al.cxx:2]: might as well update the dates for the new files

File: FastCaloSimAnalyzer/FastCaloGpu/src/GeoLoadGpu_al.cxx [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+  pImpl->reg_vec_host.push_back(cells);
+  return (long long*)alpaka::getPtrNative(cells);
+}
+
+bool GeoLoadGpu::LoadGpu_al() {
+  if ( !m_cells || m_ncells == 0 ) {
+    std::cout << "Geometry is empty " << std::endl;
+    return false;
+  }
+
```
Comment (cgleggett) [inline FastCaloSimAnalyzer/FastCaloGpu/src/GeoLoadGpu_al.cxx:270]: can we printout what backend device it's running on?

File: FastCaloSimAnalyzer/FastCaloGpu/src/CaloGpuGeneral.cxx [hunk 3]
```diff
@@ -70,6 +71,8 @@ void* CaloGpuGeneral::Rand4Hits_init( long long maxhits, unsigned short maxbin,
 void CaloGpuGeneral::Rand4Hits_finish( void* rd4h ) {
   #ifdef USE_STDPAR
   CaloGpuGeneral_stdpar::Rand4Hits_finish( rd4h );
+  #elif defined (USE_ALPAKA)
```
Comment (cgleggett) [inline FastCaloSimAnalyzer/FastCaloGpu/src/CaloGpuGeneral.cxx:74]: stupid github won't let me add a comment on a line you haven't touched... so on line 63, need to add another #elseif block for USE_ALPAKA

File: FastCaloSimAnalyzer/FastCaloGpu/src/GeoLoadGpu_al.cxx (hunk 2 above)
Comment (tsulaiav) [reply FastCaloSimAnalyzer/FastCaloGpu/src/GeoLoadGpu_al.cxx:270]: We have a bunch of printouts in this function. Do you want to add device/backend info to all of them or only to this one?

File: FastCaloSimAnalyzer/FastCaloGpu/src/CaloGpuGeneral.cxx (hunk 3 above)
Comment (tsulaiav) [reply FastCaloSimAnalyzer/FastCaloGpu/src/CaloGpuGeneral.cxx:76]: Fixed

File: FastCaloSimAnalyzer/FastCaloGpu/src/GeoLoadGpu_al.cxx (hunk 2 above)
Comment (cgleggett) [reply FastCaloSimAnalyzer/FastCaloGpu/src/GeoLoadGpu_al.cxx:270]: only this one - have a look at the other implementations and see if you can reproduce the syntax, eg:
`Executing on GPU: NVIDIA RTX A6000`

Comment (tsulaiav) [reply FastCaloSimAnalyzer/FastCaloGpu/src/GeoLoadGpu_al.cxx:270]: OK, fixed. Also applied a couple of fixes to `CMakeLists.txt`

**Pull request merged**